apps/backup-and-flush/backup_and_flush.py

```diff
@@ -0,0 +1,221 @@
+
+
+def check_wal_flushed(client: weaviate.Client, class_name: str, timestamp: datetime.datetime):
+    # Mapped volume to have access to Weaviate's data directory from container
+    # if running this script locally replace with ../apps/weaviate/data
+    weaviate_dir = "/data"
+
+    shard_name = get_shard_name(client, class_name)
+    objects_dir = f"{weaviate_dir}/{class_name.lower()}/{shard_name}/lsm/objects"
+    files = os.listdir(objects_dir)
+    for file in files:
+        if file.endswith(".wal"):
+            file_path = os.path.join(objects_dir, file)
+            file_size = os.path.getsize(file_path)
+            file_size_mb = file_size / (1024 * 1024)
+            creation_time = os.path.getctime(file_path)
+            dt_creation_time = datetime.datetime.fromtimestamp(creation_time)
+            print(f"File: {file}, Size: {file_size_mb} bytes, Created: {dt_creation_time}")
+            delay = dt_creation_time - timestamp
+            if file_size_mb == 0.0:
+                logger.info(
+                    f"WAL file was flushed after {default_flush_timeout} seconds of its creation."
+                )
+            else:
+                if delay > datetime.timedelta(seconds=default_flush_timeout):
+                    fatal(
+                        f"WAL file {file_path} was not flushed since more than {default_flush_timeout} seconds. WAL is {delay.total_seconds()} seconds old and has a size of {file_size_mb} MB"
+                    )
+                else:
+                    fatal(
+                        f"Sleeping for {default_flush_timeout} seconds was not enough to flush the WAL file {file_path}. WAL is {delay.total_seconds()} seconds old and has a size of {file_size_mb} MB"
+                    )
+
+
+client = weaviate.Client("http://localhost:8080")
+
+backup_name = f"{int(datetime.datetime.now().timestamp())}"
+
+class_names = ["Vector"]
+expected_count = 100_000
+default_flush_timeout = 60
+
+logger.info(f"Step 0, reset everything, import schema")
+reset_schema(client, class_names)
+
+logger.info(f"Step 1, import objects across {len(class_names)} classes")
+for class_name in class_names:
+    load_records(
+        client,
+        class_name,
+        start=0,
+        end=50_000,
+        stage="stage_1",
+        all_classes=class_names,
+    )
+
+logger.info("Step 2, create backup of current instance including all classes")
+create_backup(client, backup_name)
+
+# Get a timestamp on when the second import starts
+start_import_timestamp = datetime.datetime.now()
+
+logger.info(f"Step 3, import second half of objects across {len(class_names)} classes")
+for class_name in class_names:
+    load_records(
+        client,
+        class_name,
+        start=50_000,
+        end=100_000,
+        stage="stage_2",
+        all_classes=class_names,
+    )
+
+
+logger.info(f"Step 4, sleep for {default_flush_timeout} seconds to allow WAL to flush")
+time.sleep(default_flush_timeout)
```

Comment:
Flush callback is triggered in intervals (sometimes even 5 s).
I would give an extra buffer for sleep (like 10s) to make sure that flush callback had a chance to assert that 60s has passed since memtable was written to and trigger flush.

Lets imagine 1st flush callback will be executed 3s after above sleep was called (its async process, so we don't know exactly when it will run). Next ones could be then triggered at 8s, 13s, 18s, 23s, 28s, 33s, 38s, 43s, 48s, 53s, 58s, 63s, etc.
Only the last one would actually find memtable eligible to be flushed, as 63s > `default_flush_timeout`, so wal will be replaced with new one after 63s.

Reply:
Makes sense. I will go ahead and update the PR and updated at the same time the WEAVIATE_VERSION, as 1.24.2 was released yesterday.
Thanks for the feedback @aliszka